src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -72,7 +72,7 @@
     public JsonParser(TypeToken<T> type) {
         this.type = type;
         this.gson = new GsonBuilder()
-                .registerTypeAdapter(Date.class, new SaltStackDateDeserializer())
+                .registerTypeAdapter(Date.class, new DateAdapter().nullSafe())
```

Comment:
As already discussed: I'm not sure if we really want this to be null safe, it would mean we accept **any** missing dates in the returned JSON, right? I would personally prefer to fail if any expected date is missing. This way we might even uncover bugs in salt-api while we can still decide to accept that a date is missing per field once it actually occurs.